code/modules/surgery/organs/augments_internal.dm

```diff
@@ -361,6 +361,11 @@
 			addtimer(CALLBACK(src, .proc/heal), 30)
 		else
 			reviving = FALSE
+			if(owner.HasDisease(new /datum/disease/critical/shock(0)) && prob(15)) //If they are no longer in crit, but have shock, and pass a 15% chance:
```

Comment:
So... the only reason this spawns a new `/datum/disease` is so that it can call `IsSame()` on it.
https://github.com/ParadiseSS13/Paradise/blob/c5e593b2bfc2444b13cc877c2e47592866be5cb1/code/datums/diseases/_MobProcs.dm#L2-L7
`IsSame()` of course, being nothing but a wrapper for `istype()`.
https://github.com/ParadiseSS13/Paradise/blob/c5e593b2bfc2444b13cc877c2e47592866be5cb1/code/datums/diseases/_disease.dm#L155-L158

Realistically it's probably out of scope of this PR, but if you could refactor `HasDisease()` to just check the type normally instead of... whatever it's doing currently, it'd only add a few extra changed lines and I'd be very grateful 🙏.

All seems to work fine otherwise, so 👍.